core/api/src/adapter.rs

```diff
@@ -239,10 +239,15 @@ where
         let storage_mpt_tree = MPTTrie::from_root(account.storage_root, Arc::clone(&self.trie_db))?;
 
         let hash: Hash = BigEndianHash::from_uint(&position);
-        storage_mpt_tree
+        let value: H256 = storage_mpt_tree
             .get(hash.as_bytes())?
-            .map(Into::into)
-            .ok_or_else(|| APIError::Adapter("Can't find this position".to_string()).into())
+            .map(|v| BigEndianHash::from_uint(&U256::decode(v).unwrap()))
+            .ok_or_else(|| {
```

Comment:
I believe `getStorage` should also return the zero value for non-existant account or storage slot.